src/main/resources/admin/tools/content-type-manager/components/component-file-handler.html

```diff
@@ -39,44 +39,60 @@
           }
         },
         async loadCty() {
-          if(this.checkBrowserCompatibility()){
-            Swal.fire({
+
+          const showSelectXmlDialog = async () => {
+            return Swal.fire({
               text: this.warnings['select-xml'],
               showCancelButton: true,
             }).then(async (result) => {
-              if (!result.isConfirmed) return;
-
-              [fileHandle] = await window.showOpenFilePicker();
-              this.fileHandle = fileHandle;
-              const file = await fileHandle.getFile();
-
-              if (file.type !== "text/xml") {
-                Swal.fire({
-                  text: this.warnings['file-not-xml'],
-                  icon: "error",
-                  showConfirmButton: false,
-                  timer: 1500,
-                });
-                this.fileHandle = null;
-                return false;
-              }
+              if(!result.isConfirmed) { return Promise.reject(); }
+            })
+          }
 
-              this.setJson(await file.text());
+          const setFileHandler = async () => {
+            [fileHandle] = await window.showOpenFilePicker();
+            this.fileHandle = fileHandle;
+            const file = await fileHandle.getFile();
+            
+            if(!file){ return Promise.reject(); }
 
-              Swal.fire({
-                iconHtml: `<img class="loading-icon" src="${app.$root._data.loadingIcon}">`,
-                customClass: {
-                  icon: "no-border",
-                },
-                showConfirmButton: false,
-                timer: 1500,
-              }).then(() => {
-                this.$store.commit("setFileName", {
-                  data: file.name.replace(".xml", ""),
-                });
+            return file;
+          }
+
+          const checkIfFileIsXml = (file) => {
+            if (file.type !== "text/xml") {
+              throw this.warnings['file-not-xml'];
+            }
+          }
+
+          const showLoadingDialog = (file) => {
+            Swal.fire({
+              iconHtml: `<img class="loading-icon" src="${app.$root._data.loadingIcon}">`,
+              customClass: { icon: "no-border" },
+              showConfirmButton: false,
+              timer: 1500,
+            }).then(() => {
+              this.$store.commit("setFileName", {
+                data: file.name.replace(".xml", ""),
               });
             });
           }
+
+          showSelectXmlDialog()
```

Comment:
Nice, does it work and the loading closes when its done with converting the xml to json and generating the dom?